### DISCUSSION TITLE
Document Level Control in Azure Cognitive Search

### DISCUSSION THREAD

**Discussion by gcarvajal1222** [microsoft/sample-app-aoai-chatGPT#670] [General]
Hello! absolutely love the couple of click deployments on this app. I was wondering about the AZURE_SEARCH_PERMITTED_GROUPS_COLUMN env variable. Are we meant to update the index with a "column" called group_ids into a collection of strings in the index? I am using the generate your own chatbot via azure open studio but do want to put document level access control. It would be great if there is a bit more explanation on how to use this env variable as I think this feature is very important :)

**Comment by hemastransform** [comment]
Hi did you able to figure this out? i  am also working on a project similar to yours, still trying to get the document level access for the documents, if you find a method can you please share??

**Reply by jack-vinitsky** [reply]
I'm looking to implement this as well.  When it says group ids, does this mean the object id of the Azure AD group(s) are what needs to be stored in the index field?

**Comment by jack-vinitsky** [comment]
> Hello! absolutely love the couple of click deployments on this app. I was wondering about the AZURE_SEARCH_PERMITTED_GROUPS_COLUMN env variable. Are we meant to update the index with a "column" called group_ids into a collection of strings in the index? I am using the generate your own chatbot via azure open studio but do want to put document level access control. It would be great if there is a bit more explanation on how to use this env variable as I think this feature is very important :)

I think you can call the column in your index whatever you want.  The name chosen needs to be stored in AZURE_SEARCH_PERMITTED_GROUPS_COLUMN environment variable.

**Reply by jack-vinitsky** [reply]
> > Hello! absolutely love the couple of click deployments on this app. I was wondering about the AZURE_SEARCH_PERMITTED_GROUPS_COLUMN env variable. Are we meant to update the index with a "column" called group_ids into a collection of strings in the index? I am using the generate your own chatbot via azure open studio but do want to put document level access control. It would be great if there is a bit more explanation on how to use this env variable as I think this feature is very important :)
> 
> I think you can call the column in your index whatever you want. The name chosen needs to be stored in AZURE_SEARCH_PERMITTED_GROUPS_COLUMN environment variable.

The field in your index I believe should look like this:

`{"name": "group_ids", "type": "Collection(Edm.String)", "filterable": true, "retrievable": false }`

I am going by this article:  https://learn.microsoft.com/en-us/azure/search/search-security-trimming-for-azure-search

I am curious if we enable this feature in the application, what happens to documents that do not have any values in this field?  Are they included the search results or do they all fall out?  If they fall out, then I suppose I would need to assign a group that contains all users in my company as a "default" value for this field during our indexing.

**Reply by jack-vinitsky** [reply]
So I configured the group_ids field in the Search Index as indicated in my prior post and set the AZURE_SEARCH PERMITTED_GROUPS_COLUMN.  I added a step to the AI SEARCH indexer (we use integrated vectorization) to store the group id of a specific security group for a set of documents and a "fake group id" for a set of documents that I want to be filtered out.

I then reindexed my documents.  Unfortunately I am still getting documents used in answers that have the fake group id.

I assume I should be storing the object id in the Azure Search Field, correct?

![image](https://github.com/user-attachments/assets/2641a6d9-a2db-4cd1-a19e-d94c897092bf)
